File: PROGRAMMERS/LEVEL1/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int cnt1 = 0;
    int cnt2 = 0;
    for (int i = 0; i < 6; i++) {
        for (int j = 0; j < 6; j++) {
            if (lottos[i] == win_nums[j]) {
                cnt1++;
                cnt2++;
            }
        }
        if (lottos[i] == 0) cnt2++;
    }

    int result = 7 - cnt2;
    if (result >= 6) result = 6;
    answer.push_back(result);

    result = 7 - cnt1;
    if (result >= 6) result = 6;
    answer.push_back(result);

    return answer;
}

File: PROGRAMMERS/LEVEL1/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int[] answer = {0,0};
        
        int rankMax = 0;
        int rankMin = 0;
        
        for(int i=0;i<6;i++) {
        	if(lottos[i]==0) {
        		rankMax++;
        		continue;
        	}
        	for(int j=0;j<6;j++) {
        		if(lottos[i]==win_nums[j]) {
        			rankMax++;
        			rankMin++;
        		}
        	}
        }
        
        int result = 7 - rankMax;
        if(result>=6) result = 6;
        answer[0] = result;
        
        result = 7 - rankMin;
        if(result>=6) result = 6;
        answer[1] = result;
        
        return answer;
    }
}